src/Common/WellKnown/MoqKnownSymbols.cs

```diff
@@ -5,14 +5,27 @@ namespace Moq.Analyzers.Common.WellKnown;
 
 internal class MoqKnownSymbols : KnownSymbols
 {
+    private readonly Lazy<ImmutableArray<IMethodSymbol>> _mockRepositoryCreate;
```

Comment:
This update needs documentation for future maintainers. Why are things being cached? What happens when symbols are updated (e.g., Moq versions are changed), is the cache still valid and safe? Reconcile that against guidance for writing Roslyn analyzers to never cache anything yourself (i.e., allow the analyzer framework to do that, it knows about the nuances, we don't and are subject to behavior changes between SDK versions)